crates/core/tedge_api/examples/heartbeat.rs

```diff
@@ -259,6 +264,10 @@ impl Handle<Heartbeat> for CriticalService {
     }
 }
 
+impl PluginDeclaration for CriticalService {
+    type HandledMessages = (Heartbeat,);
+}
```

Comment:
I miss something here. Is this `HandledMessages` a tuple type? If so, how a tuple type is converted into the enum type of the messages actually sent to the plugin?

I feel that a more appropriate example would have two types of messages.
Would this be something along these lines? 
```
impl PluginDeclaration for MyPlugin {
    type HandledMessages = (A,B);
}
impl Handle<A> for MyPlugin {
  // ...
}
impl Handle<B> for MyPlugin {
  // ...
}
```

Reply:
I am not sure which part you addressing, but there is no enum at any point. The trait doing the heavy lifting is `DoesHandle` and the `impl_does_handle_tuple` macro in `src/plugin.rs`.

I don't mind adding another message type to the heartbeat example, but don't consider it necessary.

Reply:
I was looking for the type grouping all the message types that can be received by a plugin: the item type of the receiver  channel of that plugin. I have my response now: this type is `Any` or more precisely `InternalMessage`  boxing some `Any` value. I was looking for a type known at compile time and specific to the plugin, hence an enum, rather that this global type with dynamic downcasts. But it's okay. It makes sense with the dynamic plugin-connectivity approach of this proposal.